hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -332,10 +347,42 @@ def from_typing_type(thing):
             # Reversible is somehow a subclass of Hashable, so we tuplize it.
             # See also the discussion at https://bugs.python.org/issue39046
             typing.Reversible: st.lists(st.integers()).map(tuple),  # type: ignore
-            typing.SupportsAbs: st.complex_numbers(),
-            typing.SupportsComplex: st.complex_numbers(),
-            typing.SupportsFloat: st.floats(),
-            typing.SupportsInt: st.floats(),
+            typing.SupportsAbs: (
+                st.booleans()
+                | st.integers()
+                | st.floats()
+                | st.complex_numbers()
+                | st.fractions()
+                | st.decimals()
+                | st.timedeltas()
+            ),
+            typing.SupportsComplex: (
+                st.booleans()
+                | st.integers()
+                | st.floats()
+                | st.complex_numbers()
+                | st.decimals()
+                | st.fractions()
+            ),
+            typing.SupportsFloat: (
+                st.booleans()
+                | st.integers()
+                | st.floats()
+                | st.decimals()
+                | st.fractions()
+                # with floats its far more annoying to capture all
+                # the magic in a regex. so we just stringify some floats
+                | st.floats().map(str)
+            ),
+            typing.SupportsInt: (
+                st.booleans()
+                | st.integers()
+                | st.floats()
+                | st.uuids()
+                | st.decimals()
+                # this generates strings that should able to be parsed into integers
+                | st.from_regex(r"^-?([1-9]\d*)|0$", fullmatch=True)
```

Comment:
This isn't quite equivalent to `st.from_regex(r"-?\d+", fullmatch=True).filter(can_cast_to_int)`, and I'd prefer the more general strategy - your regex never permits non-ascii leading digits, for example.

Reply:
ah yeah... ok point taken. will do